### PR TITLE
PWX-38544 | Improvements for generic helm 

### PR DESCRIPTION
**Pull request by hitesh-wani-px**
**What this PR does / why we need it**:
**Improvements**
1. Update helm notes  which has incorrect links and readme file
2. Remove PX_SECRETS_NAMESPACE env variable 
3. Following PX components should be enabled by default in the values.yaml
    1. autopilot
    2. CSI
    3. internalKvdb
   
**Which issue(s) this PR fixes** (optional)
Closes #https://purestorage.atlassian.net/browse/PWX-38544

**Special notes for your reviewer**:

### CODE REVIEW

**Comment by nikita-bhatia** [comment]
@hitesh-wani-px please re-check once if we really want to remove PX_SECRETS_NAMESPACE  if it was part of helm already, as I don't see it is deprecated.

File: charts/portworx/templates/storage-cluster.yaml [hunk 1]
```diff
@@ -292,6 +292,9 @@ spec:
     {{- end }}
   {{- end }}
 
+  {{- $isLicenseSecretAdded := ne $licenseSecret "none" }}
+  {{- $shouldRenderEnv := or (ne $envVars "none") .Values.envs $AKSInstall $OKEInstall $isLicenseSecretAdded }}
```
Comment (diptiranjanpx) [inline charts/portworx/templates/storage-cluster.yaml:296]: Is the following limited to AKS and OKE?

**Review by diptiranjanpx — APPROVED** [review]

File: charts/portworx/templates/storage-cluster.yaml (hunk 1 above)
Comment (hitesh-wani-px) [reply charts/portworx/templates/storage-cluster.yaml:297]: Yes some env var are that needs to be included are related to AKS and OKE only

**Comment by hitesh-wani-px** [comment]
> PX_SECRETS_NAMESPACE



> @hitesh-wani-px please re-check once if we really want to remove PX_SECRETS_NAMESPACE if it was part of helm already, as I don't see it is deprecated.

The PX_SECRETS_NAMESPACE environment variable is currently being set by default, which should not be the case. It is not added as a default environment variable in the spec generation as well. If users require this variable, they can add it later using the env field. Removing this variable will not affect the installation process. To align with the spec generation, we should remove it as a default.

**Review by nikita-bhatia — APPROVED** [review]
LGTM

**Pull request merged**